### PR TITLE
Download xml file & change name of the file

### PR DESCRIPTION
**Pull request by wawax**
Factorize code to display FI infos on assets
Rename XML files stored on the filesystem : add a .xml suffix
Add a download XML function for network equipments and printers

### CODE REVIEW

File: inc/item.class.php [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      return true;
+   }
+
+   /**
+   * Add more info to the tab if needed
+   * @since 9.2+2.0
+   *
+   * @param array $options display options
+   */
+   function addMoreInfos($options = []) {
```
Comment (trasher) [inline inc/item.class.php]: Why is this method empty?

File: inc/item.class.php [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+   }
+
+   /**
+   * Get a FI class instance representing an itemtype
+   * @since 9.2+2.0
+   *
+   * @param array $options display options
+   */
+   static function getFIItemClassInstance($itemtype) {
```
Comment (trasher) [inline inc/item.class.php]: Maybe should we add a message in logs when it is called with an unknown item type?

File: inc/networkequipment.class.php [hunk 3]
```diff
@@ -90,24 +92,26 @@ static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtempl
...
-                 array('target'=>$CFG_GLPI['root_doc'].
-                                    '/plugins/fusioninventory/front/switch_info.form.php'));
+            case 'NetworkPort$1':
+               $pfNetworkEquipment->showExtendedInfos($item, $options);
+               break;
          }
-         return TRUE;
       }
-      return FALSE;
+      return false;
```
Comment (trasher) [inline inc/networkequipment.class.php]: Method will always return `false`; is that intended?

File: inc/networkequipment.class.php [hunk 4]
```diff
@@ -153,11 +156,11 @@ function showForm(CommonDBTM $item, $options=array()) {
          }
       }
 
-      if (isset($_POST['displaysnmpinfo'])) {
-         $this->showNetworkEquipmentInformation($id, $options);
-         return;
-      }
-//$_SESSION['plugin_fusioninventory_networkportview'] = 'glpi';
+      //if (isset($_POST['displaysnmpinfo'])) {
```
Comment (trasher) [inline inc/networkequipment.class.php]: Please drop commented code

File: inc/networkequipment.class.php [hunk 5]
```diff
@@ -431,8 +432,8 @@ class='tab_bg_1_2'>".$icon.$item->getLink(1);
                   $link1 = $item->getLink(1);
                   $link = str_replace($item->getName(0), $NetworkPort->fields["mac"],
                                       $item->getLink());
-//                  $link2 = str_replace($item->getName(0), $NetworkPort->fields["ip"],
-//                                       $item->getLink());
+                  //                  $link2 = str_replace($item->getName(0), $NetworkPort->fields["ip"],
```
Comment (trasher) [inline inc/networkequipment.class.php]: Please drop commented code

File: inc/networkequipment.class.php [hunk 6]
```diff
@@ -489,14 +489,14 @@ function updateNetworkInfo($id, $configsecurities_id, $sysdescr) {
...
 
-
    /**
     * Display network equipment extended information
     *
     * @global object $DB
     * @param object $item
     * @param array $options
     */
+    /*
```
Comment (trasher) [inline inc/networkequipment.class.php]: Should be dropped

File: inc/item.class.php [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+
+   /**
+   * The itemtype we are processing
+   */
+   public $itemtype  = '';
+
+   /**
+   * Get automatic inventory info for a computer
+   * @since 9.1+1.2
+   * @param $items_id the item ID to look for
```
Comment (trasher) [inline inc/item.class.php]: Param type is missing

File: inc/item.class.php [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+         return $this->fields;
+      } else {
+         return [];
+      }
+   }
+
+   /**
+   * Form to download the XML inventory file
+   *
+   * @param $items_id the item ID to look for
```
Comment (trasher) [inline inc/item.class.php]: Param type is missing
Return is missing as well

File: inc/item.class.php [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+         echo "&nbsp;$message</a>";
+         echo "</th></tr></table>";
+      }
+   }
+
+   /**
+    * Display form
+    *
+    * @param object $item CommonDBTM instance
+    * @param array $options
```
Comment (trasher) [inline inc/item.class.php]: Param description is missing. Params/types/description should be indented the same

File: inc/networkequipment.class.php (hunk 6 above)
Comment (wawax) [reply inc/networkequipment.class.php]: done

File: inc/networkequipment.class.php (hunk 5 above)
Comment (wawax) [reply inc/networkequipment.class.php]: done

File: inc/networkequipment.class.php (hunk 4 above)
Comment (wawax) [reply inc/networkequipment.class.php]: done

File: inc/networkequipment.class.php (hunk 3 above)
Comment (wawax) [reply inc/networkequipment.class.php]: no it should return true, as for other objects

File: inc/item.class.php (hunk 1 above)
Comment (wawax) [reply inc/item.class.php]: this method enables a child class to add more data to display. By default it's empty, PluginFusioninventoryNetworkEquipment overrides the method

File: inc/item.class.php (hunk 9 above)
Comment (wawax) [reply inc/item.class.php]: done

File: inc/item.class.php (hunk 8 above)
Comment (wawax) [reply inc/item.class.php]: done

File: inc/item.class.php (hunk 7 above)
Comment (wawax) [reply inc/item.class.php]: done

**Pull request merged**